File: SuperMovie_server/src/api/Schedule/AddSchedule.cs
namespace SuperMovie.Server.Api.Schedule;

using WebSocketSharp;
using WebSocketSharp.Server;

public struct AddScheduleReq
{
    public long ScheduleFilmId;
    public DateTime ScheduleStartTime;
    public DateTime ScheduleEndTime;
}

public struct AddScheduleRsp
{
    public bool Ok;
    public long ScheduleId;
}

//api : add_schedule
public class AddSchedule : WebSocketBehavior
{
}

File: movieui/src/scripts/ws/Schedule/addSchedule.ts
import {wsUrlRoot} from "@/scripts/ws/meta"
import {recvMsg, reqStringify, rspParse, sendMsg} from "@/scripts/ws/helper"

export {addSchedule}

const conn = new WebSocket(`${wsUrlRoot}/add_schedule`)

type AddScheduleReq =
    {
        ScheduleFilmId: bigint
        ScheduleStartTime: Date
        ScheduleEndTime: Date
    }

type AddScheduleRsp =
    {
        Ok: boolean
        ScheduleId: bigint
    }

async function addSchedule(req: AddScheduleReq) {

    const task = recvMsg(conn)
    sendMsg(conn, reqStringify(req)).then()
    const msg = await task

    return <AddScheduleRsp>rspParse(msg)
}